dockstore-common/src/main/java/io/dockstore/common/metrics/MetricsDataS3Client.java

```diff
@@ -98,6 +102,27 @@ public static String generateKey(String toolId, String versionName, String platf
         return String.join("/", pathList);
     }
 
+    public boolean doesExecutionExistInS3(String toolId, String versionName, String platform, String executionId) {
+        final String fileName = S3ClientHelper.appendJsonFileTypeToFileName(executionId);
+        String s3Key = MetricsDataS3Client.generateKey(toolId, versionName, platform, fileName);
```

Comment:
Will the model always be that "platform" and the reporter are the same party?  If yes, that's ok.  I ask because one can imagine a scenario in which there's some third party platform X which doesn't submit metrics, and instead, the _users_ of X are submitting metrics of the runs, in which case there'd need to be measures in place to ensure that the IDs from the different sources didn't collide.  But it's ok if we don't support the latter.